docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,354 @@
+  "This documentation cover methods related to GraphQL exposed by
+  `@front-commerce/core`."
+---
+
+<p>{frontMatter.description}</p>
+
+## `createGraphQLModule`
+
+Creates a GraphQL module and injects it into Front-Commerce's GraphQL server.
+
+```ts
+import { createGraphQLModule } from "@front-commerce/core/graphql";
+import AcmeModule from "./acmeModule";
+
+export default createGraphQLModule({
+  namespace: "Acme",
+  modules: [AcmeModule],
+});
+```
+
+| Argument    | Type                   | Description                         |
+| ----------- | ---------------------- | ----------------------------------- |
+| `namespace` | `string`               | The namespace of the GraphQL module |
+| `modules`   | `Array<GraphQLModule>` | The modules to inject in namespace  |
+
+## `readGraphQLModules`
+
+Reads and parses GraphQLModules generated by codegen.
+
+_Example_
+
+```ts
+import { readGraphQLModules } from "@front-commerce/core/graphql";
+
+const generatedSchemaModules = await readGraphQLModules();
+```
+
+## `schema`
+
+Reads the generated GraphQL schema.
+
+```ts
+import { schema } from "@front-commerce/core/graphql";
+
+const acmeSchema = await schema();
+```
+
+## `createGraphQLApi`
+
+Creates a GraphQL method to executes GraphQL queries and mutations.
+
+```ts
+import { createGraphQLApi } from "@front-commerce/core/graphql";
+import { AcmeQuery } from "~/graphql/graphql";
+
+const AcmeGraphQLApi = createGraphQLApi(request, user, config);
+
+const response = await AcmeGraphQLApi(AcmeQuery);
+```
+
+Arguments:
+
+| Argument  | Type      | Description                   |
+| --------- | --------- | ----------------------------- |
+| `request` | `Request` | The request object            |
+| `user`    | `User`    | The user context              |
+| `config`  | `Config`  | The application configuration |
+
+## `makeErrorLoggerInterceptor`
+
+Intercepts and log errors.
+
+```ts
+import { makeErrorLoggerInterceptor } from "@front-commerce/core/graphql";
+
+const errorLoggerInterceptor = makeErrorLoggerInterceptor({
+  isMaintenance: (error) =>
+    error.response &&
+    (error.response.status === 503 ||
+      (error.response.status === 500 &&
+        /maintenance mode .* enabled/.test(error.response.data))),
+  ignoreForbiddenErrors: ignoreForbiddenErrors,
+});
+```
+
+Arguments:
+
+| Argument                | Type       | Description                                                                                                 |
+| ----------------------- | ---------- | ----------------------------------------------------------------------------------------------------------- |
+| `isMaintenance`         | `function` | a function that recieve error as parameter and return a `boolean` to tell if website is in maintenance mode |
+| `ignoreForbiddenErrors` | `boolean`  | a boolean to tell if forbidden errors                                                                       |
+| should be ignored       |
+
+## `limitRateByClientIp`
```

Comment:
**issue:** removed in FC-1558

**suggestion:** remove this section

Reply:
removed